### PR TITLE
[enh] Simplify error management

### PR DESCRIPTION
**Pull request by irina11y**
## The problem

Utilisation trop complexe. Fait perdre du temps de dev.

## Solution

Supression des code d'erreur "errno et de l'internationalisation "m18n.n"

### CODE REVIEW

**Comment by Josue-T** [comment]
As I'm working now on some new implementation, it could be good to know what will happen with this PR.

**Comment by zamentur** [comment]
For information, this PR depends of a change on Moulinette.

I am ok on principle, but we need to test that correctly to avoid some fail due to typo issue...

**Comment by irina11y** [comment]
Depends of  https://github.com/YunoHost/moulinette/pull/180

File: src/yunohost/user.py [hunk 1]
```diff
@@ -342,8 +337,7 @@ def user_update(operation_logger, auth, username, firstname=None, lastname=None,
         for mail in add_mailalias:
             auth.validate_uniqueness({'mail': mail})
             if mail[mail.find('@') + 1:] not in domains:
-                raise MoulinetteError(errno.EINVAL,
-                                      m18n.n('mail_domain_unknown',
+                raise MoulinetteError(m18n.n('mail_domain_unknown',
```
Comment (zamentur) [inline src/yunohost/user.py]: idem

File: src/yunohost/utils/yunopaste.py [hunk 2]
```diff
@@ -13,17 +13,14 @@ def yunopaste(data):
     try:
         r = requests.post("%s/documents" % paste_server, data=data, timeout=30)
     except Exception as e:
-        raise MoulinetteError(errno.EIO,
-                              "Something wrong happened while trying to paste data on paste.yunohost.org : %s" % str(e))
+        raise MoulinetteError("Something wrong happened while trying to paste data on paste.yunohost.org : %s" % str(e))
```
Comment (zamentur) [inline src/yunohost/utils/yunopaste.py]: Those string was not translated, so we need to translate it or find a solution

File: src/yunohost/utils/yunopaste.py [hunk 3]
```diff
@@ -13,17 +13,14 @@ def yunopaste(data):
...
 
     if r.status_code != 200:
-        raise MoulinetteError(errno.EIO,
-                              "Something wrong happened while trying to paste data on paste.yunohost.org : %s, %s" % (r.status_code, r.text))
+        raise MoulinetteError("Something wrong happened while trying to paste data on paste.yunohost.org : %s, %s" % (r.status_code, r.text))
```
Comment (zamentur) [inline src/yunohost/utils/yunopaste.py]: idem

File: src/yunohost/utils/yunopaste.py [hunk 4]
```diff
@@ -13,17 +13,14 @@ def yunopaste(data):
...
 
     try:
         url = json.loads(r.text)["key"]
     except:
-        raise MoulinetteError(errno.EIO,
-                              "Uhoh, couldn't parse the answer from paste.yunohost.org : %s" % r.text)
+        raise MoulinetteError("Uhoh, couldn't parse the answer from paste.yunohost.org : %s" % r.text)
```
Comment (zamentur) [inline src/yunohost/utils/yunopaste.py]: idem

**Comment by alexAubin** [comment]
This should now be ready for final review (and careful final re-reading to spot any typo that would have been missed so far).

(The branch has been rebased to current stretch-unstable and misc fixes have been implemented, tested the whole thing with the unit/functional tests)

File: src/yunohost/utils/error.py [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+    You should have received a copy of the GNU Affero General Public License
+    along with this program; if not, see http://www.gnu.org/licenses
+
+"""
+
+from moulinette.core import MoulinetteError
+from moulinette import m18n
+
+class YunohostError(MoulinetteError):
```
Comment (Psycojoker) [inline src/yunohost/utils/error.py]: What is the idea of doing a specific class for YunoHost errors? I don't see any justification nowhere and that should be in a comment for when we'll ask ourself this not obvious question in 6 month or more.

Comment (alexAubin) [reply src/yunohost/utils/error.py]: The idea is that m18n.g got factorized in MoulinetteError. But in the context of YunoHost, we shall use m18n.n, hence the introduction of YunohostError.

(I added a comment https://github.com/YunoHost/yunohost/pull/574/commits/46972788b2f4c3b3ac79e2d2fb9b8dd6a3834148 )

File: src/yunohost/utils/error.py [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    """Yunohost base exception"""
+    def __init__(self, key, __raw_msg__=False, *args, **kwargs):
+        if __raw_msg__:
```
Comment (Psycojoker) [inline src/yunohost/utils/error.py]: Should be `raw_msg` and not `__raw_msg__`, dunderscores are only used for privates methodes.

Also I think that `m18n.n` will return the raw string if no translations are found so I'm not sure that this is needed.

**Comment by Josue-T** [comment]
When this PR, is merged, it mean that all other PR will need to be fixed ?

**Comment by alexAubin** [comment]
>When this PR, is merged, it mean that all other PR will need to be fixed ?

Yes, essentially you need to : 
- rebase
- replace all `MoulinetteError(m18n.n('some_key'))` by `YunohostError('some_key')`

**Comment by Josue-T** [comment]
Yes, on my side it's ok but it just imply that some PR might be never updated and so never merged...

**Comment by Psycojoker** [comment]
> Yes, on my side it's ok but it just imply that some PR might be never updated and so never merged...

If a PR is never updated it will never been merged, this PR being done or not :/

And for this kind of small changes we can do the updates ourselves.

**Pull request merged**